packages/next/src/server/app-render/app-render.tsx

```diff
@@ -1223,20 +1223,21 @@ async function renderToStream(
     renderOpts.page
   )
 
-  const reactServerErrorsByDigest: Map<string, DigestedError> = new Map()
+  // Including both RSC errors and SSR errors
+  const renderErrorsByDigest: Map<string, DigestedError> = new Map()
   const silenceLogger = false
   const serverComponentsErrorHandler = createHTMLReactServerErrorHandler(
     !!renderOpts.dev,
     !!renderOpts.nextExport,
-    reactServerErrorsByDigest,
+    renderErrorsByDigest,
     silenceLogger,
     // RSC rendering error will report as SSR error
     // @TODO we should report RSC errors where they happen for instrumentation purposes
     // and should omit the error reporter in the SSR layer instead
     undefined
   )
-  function onServerRenderError(err: DigestedError) {
-    const renderSource = reactServerErrorsByDigest.has(err.digest)
+  function onServerRenderError(err: DigestedError, isRSCError: boolean) {
+    const renderSource = isRSCError
```

Comment:
I'd refactor this to have two functions. one for RSC errors and one for SSR errors. Then pass both to create...ErrorHandler and then call one if the digest is in the RSC error map and call the other if it is a novel error